contracts/interchain-token-service/src/shared.rs

```diff
@@ -1,35 +1,36 @@
 use core::fmt;
-use std::ops::Deref;
 
 use axelar_wasm_std::IntoContractError;
 use cosmwasm_schema::cw_serde;
 use thiserror::Error;
 
 #[derive(Error, Debug, PartialEq, Eq, IntoContractError)]
 pub enum Error {
-    #[error("invalid number of bits {0}. Must be between 1 and 256")]
+    #[error("number of bits must be between 1 and 256 (given: {0})")]
     InvalidNumberOfBits(u32),
 }
 
 #[cw_serde]
-#[serde(try_from = "u32")]
-pub struct NumBits(u32);
+#[derive(PartialOrd)]
+#[serde(try_from = "u32", into = "u32")]
+pub struct NumBits(u8);
```

Comment:
Migration will be added in a separate PR?

Reply:
We thought about doing this but it was potentially confusing. You'll need to round up when exposing this in a query to avoid backwards compatibility